src/main/kotlin/org/wfanet/measurement/dataprovider/fake/FakeDataProvider.kt

```diff
@@ -0,0 +1,136 @@
+// Copyright 2021 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.wfanet.measurement.dataprovider.fake
+
+import com.google.protobuf.ByteString
+import java.util.logging.Logger
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import org.wfanet.measurement.api.v1alpha.CombinedPublicKey
+import org.wfanet.measurement.api.v1alpha.ElGamalPublicKey
+import org.wfanet.measurement.api.v1alpha.GetCombinedPublicKeyRequest
+import org.wfanet.measurement.api.v1alpha.ListMetricRequisitionsRequest
+import org.wfanet.measurement.api.v1alpha.MetricRequisition
+import org.wfanet.measurement.api.v1alpha.PublisherDataGrpcKt.PublisherDataCoroutineStub
+import org.wfanet.measurement.api.v1alpha.UploadMetricValueRequest
+import org.wfanet.measurement.common.asBufferedFlow
+import org.wfanet.measurement.common.identity.ExternalId
+import org.wfanet.measurement.common.logAndSuppressExceptionSuspend
+import org.wfanet.measurement.common.throttler.Throttler
+
+/** Size of chunks in each streamed message for uploading sketches. */
+private const val DEFAULT_STREAM_BYTE_BUFFER_SIZE: Int = 1024 * 32 // 32 KiB
+
+/**
+ * Implements a basic Data Provider that listens for [MetricRequisition]s and fulfills them.
+ *
+ * TODO: when upgraded to v2alpha API, this should also accept a SketchConfigsCoroutineStub.
```

Comment:
A question, how do we tell what API version this class is currently using? the api has a v2alpha in the code path, but how do we tell other components? e.g., those daemons.

I guess upgrading from using v1alpha api to v2alpha would need significant code change in some cases, do we just modify the some code or create another version?